qiskit_experiments/composite/composite_experiment.py

```diff
@@ -63,6 +66,23 @@ def _add_job_metadata(self, experiment_data, job, **run_options):
         # Add sub-experiment options
         for i in range(self.num_experiments):
             sub_exp = self.component_experiment(i)
+
+            # Construct default transpile options with initial layout
+            default_transpile_options = Options(**sub_exp._default_transpile_options().__dict__)
+            default_transpile_options.update_options(
+                **{"initial_layout": sub_exp.transpile_options.get("initial_layout")}
+            )
```

Comment:
These lines reflect something that's maybe not done correctly in `BaseExperiment`. In `BaseExperiment`, the default transpile options don't include the initial layout, which is added to the non-default transpile options in `BaseExperiment.__init__`. Perhaps the initial layout could be part of the default transpile options.

Reply:
Agreed, that makes a lot more sense.

Reply:
I agree with @yaelbh that this looks like an issue with how layout is handled in base experiment. One solution would be to remove initial layout from the `BaseExperiment.__init__` entirely and only add it to the transpile options just before calling transpile in `run`:

```python
# Generate and transpile circuits
transpile_opts = self.transpile_options.__dict__
transpile_opts['initial_layout'] = list(self._physical_qubits)
circuits = transpile(self.circuits(backend), backend, **transpile_opts)
```
Then no changes should be needed for composite experiment with regards to the layout (and it would be optional to leave the current exception if you try and set initial_layout in transpile options since it is always going to be overriden anyway).